src/tools/erasure-code/ceph-erasure-code-tool.cc

```diff
@@ -0,0 +1,253 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "include/buffer.h"
+#include "include/stringify.h"
+#include "common/ceph_argparse.h"
+#include "common/config_proxy.h"
+#include "common/errno.h"
+#include "erasure-code/ErasureCode.h"
+#include "erasure-code/ErasureCodePlugin.h"
+#include "global/global_context.h"
+#include "global/global_init.h"
+#include "osd/ECUtil.h"
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string.hpp>
+
+std::vector<std::string> display_params = {
+  "chunk_count", "data_chunk_count", "coding_chunk_count"
+};
+
+void usage(const std::string message, ostream &out) {
+  if (!message.empty()) {
+    out << message << std::endl;
+    out << "" << std::endl;
+  }
+  out << "usage: ceph-erasure-code-tool test-plugin-exists <plugin>" << std::endl;
```

Comment:
i'm inclined to using named options instead of positional argument, and to use `boost::program_options` .

Reply:
I have a different opinion here. Options are good for optional arguments (as the name suggest), while all parameters are mandatory here, and positional arguments feet better -- the command line is much shorter.

Reply:
thanks. it's a new angle for me to look at the command line options. agreed. i thought it'd be better to have names for options. but yeah, your opinion makes sense to me as well.